# HG changeset patch
# User Rik <rik@octave.org>
# Date 1514393981 28800
#      Wed Dec 27 08:59:41 2017 -0800
# Node ID 42d099b841aacf352bc2f4f1678acd9c7a22e861
# Parent  19d63669d174098b7ec7ddfb4bb13c255dc82f19
erase.m: Add warning if a character array is used for PTN.

* erase.m: Add warning when PTN is a char array.  Also add BIST
test for new behavior.

* warning_ids.m: Document new warning ID "Octave:erase:chararray".

diff --git a/scripts/help/warning_ids.m b/scripts/help/warning_ids.m
--- a/scripts/help/warning_ids.m
+++ b/scripts/help/warning_ids.m
@@ -165,16 +165,23 @@
 ##
 ## @item Octave:eigs:UnconvergedEigenvalues
 ## If the @code{Octave:eigs:UnconvergedEigenvalues} warning is enabled then
 ## the eigs function will issue a warning if the number of calculated
 ## eigenvalues is less than the number of requested eigenvalues.
 ## By default, the @code{Octave:eigs:UnconvergedEigenvalues} warning is
 ## enabled.
 ##
+## @item Octave:erase:chararray
+## If the @code{Octave:erase:chararray} warning is enabled then the erase
+## function will issue a warning if the input pattern is a character array
+## rather than a string or cell array of strings.
+## By default, the @code{Octave:eigs:UnconvergedEigenvalues} warning is
+## enabled.
+##
 ## @item Octave:fopen-file-in-path
 ## By default, the @code{Octave:fopen-file-in-path} warning is enabled.
 ##
 ## @item Octave:function-name-clash
 ## If the @code{Octave:function-name-clash} warning is enabled, a
 ## warning is issued when Octave finds that the name of a function
 ## defined in a function file differs from the name of the file.  (If
 ## the names disagree, the name declared inside the file is ignored.)
diff --git a/scripts/strings/erase.m b/scripts/strings/erase.m
--- a/scripts/strings/erase.m
+++ b/scripts/strings/erase.m
@@ -75,16 +75,18 @@ function newstr = erase (str, ptn)
       ischarmatrix = true;
     endif
   elseif (! iscellstr (str))
     error ("erase: STR must be a string or cell array of strings");
   endif
 
   if (ischar (ptn)) 
     if (rows (ptn) > 1)
+      warning ("Octave:erase:chararray",
+               "erase: using character array for PTN is not recommended, consider cell array of strings instead");
       ## Convert to cell.  Can't use cellstr which trims spaces.
       ptn = mat2cell (ptn, ones (1, rows (ptn)));
     endif
   elseif (! iscellstr (ptn))
     error ("erase: PTN must be a string or cell array of strings");
   endif
 
   nptn = ifelse (ischar (ptn), 1, numel (ptn));
@@ -132,8 +134,10 @@ endfunction
 %!assert (erase ({'Hello World t '}, {'ld '; 'o '}), {'HellWort '})
 
 ## Test input validation
 %!error erase ()
 %!error erase ("a")
 %!error erase ("a", "b", "c")
 %!error <STR must be a string> erase ([1], "foo") 
 %!error <PTN must be a string> erase ("foo", [1]) 
+%!warning <using character array for PTN is not recommended>
+%! erase ("a", ["a";"b"]);
